### PR TITLE
Support https with server certificate

### LINKED ISSUE #244: Fetch need a https fetch
I think we should provide a https fetch that accept ca cert file to query https server's info

### PR DESCRIPTION
**Pull request by nolanmar511**
Fixes #244

### CODE REVIEW

**Review by aalexand — COMMENTED** [review]
I think we should rework the "Fetching profiles" section to organically (i.e. throughout all the text where applicable, not just as a separate paragraph) incorporate the documentation of the support for the profile fetch from TLS-protected server endpoints. Perhaps mention https://docs.docker.com/engine/security/https/ at some point to point to the resemblance of the flag names, server setup and the overall flow.

**Comment by nolanmar511** [comment]
Rebased and updated documentation.
PTAL

File: doc/README.md [hunk 1]
```diff
@@ -270,6 +270,18 @@ wait for the profile.
   profile over http. If not specified, pprof will use heuristics to determine a
   reasonable timeout.
 
+pprof also accepts options which allow a user to specify TLS certificates to
```
Comment (aalexand) [inline doc/README.md]: Could you rework the first paragraph in this section to mention this capability. Also, can we mention https+insecure thing as well? Basically I think the first paragraph needs a bit of holistic update to make it more of an overview with subsequent paragraphs diving into the details.

**Review by aalexand — COMMENTED** [review]
Just one comment.

File: doc/README.md (hunk 1 above)
Comment (nolanmar511) [reply doc/README.md]: I made a very small change to the first paragraph, but I'm having a hard time figuring this documentation -- I think I might not understand http protocols and pprof's use cases well enough right now to make this clear.

Comment (aalexand) [reply doc/README.md]: "https+insecure" is not a protocol, it's a hack to disable the server's certificate validation on the client side.

Comment (nolanmar511) [reply doc/README.md]: Done.

**Comment by nolanmar511** [comment]
PTAL

**Review by aalexand — APPROVED** [review]

**Review by alhajialiumar2019-max — COMMENTED** [review]
Please review this message, make any necessary corrections or improvements, and sign it upon completion.

**Pull request merged**